webrtc/videoroom/lib/videoroom/pipeline.ex

```diff
@@ -223,18 +246,24 @@ defmodule VideoRoom.Pipeline do
         _ctx,
         state
       ) do
-    participants =
-      state.endpoints
-      |> Enum.map(fn {_, %Endpoint{inbound_tracks: tracks, ctx: ctx}} ->
-        %{
-          display_name: ctx.display_name,
-          muted_audio: ctx.muted_audio,
-          muted_video: ctx.muted_video,
-          mids: Map.keys(tracks)
-        }
-      end)
+    with endpoint = %Endpoint{} <- state.endpoints[peer_pid] do
+      participants =
+        state.endpoints
+        |> Enum.map(fn {_, %Endpoint{inbound_tracks: tracks, ctx: ctx}} ->
+          %{
+            id: ctx.participant_id,
+            display_name: ctx.display_name,
+            muted_audio: ctx.muted_audio,
+            muted_video: ctx.muted_video,
+            mids: Map.keys(tracks)
+          }
+        end)
+
+      user_id = endpoint.ctx.participant_id
+
+      send(peer_pid, {:signal, message, participants, user_id})
+    end
```

Comment:
Why do we use `with` here? Also function in Enum.map can use get_participant_data

Reply:
This `with` is necessary to avoid raise condition when many peers are leaving room at this same time, e.g. when somebody with active screensharing leaves room